Source/PartnerCenter.CustomerPortal/BusinessLogic/Commerce/PaymentGateways/PayUGateway.cs

```diff
@@ -0,0 +1,437 @@
+        /// Generate hash. 
+        /// </summary>
+        /// <param name="text">hash string.</param>
+        /// <returns>return string</returns>
+        public string Generatehash512(string text)
+        {
+            byte[] message = Encoding.UTF8.GetBytes(text);
+
+            UnicodeEncoding ue = new UnicodeEncoding();
+            byte[] hashValue;
+            System.Security.Cryptography.SHA512Managed hashString = new System.Security.Cryptography.SHA512Managed();
+            string hex = string.Empty;
+            hashValue = hashString.ComputeHash(message);
+            foreach (byte x in hashValue)
+            {
+                hex += string.Format("{0:x2}", x);
+            }
+
+            return hex;
+        }
+
+        /// <summary>
+        /// Generate transaction id. 
+        /// </summary>
+        /// <returns>return string</returns>
+        public string Generatetxnid()
+        {
+            Random rnd = new Random();
+            string strHash = this.Generatehash512(rnd.ToString() + DateTime.Now);
+            string txnid1 = strHash.ToString().Substring(0, 20);
+            return txnid1;
+        }
+
+        /// <summary>
+        /// Retrieves the order details maintained for the payment gateway.  
+        /// </summary>
+        /// <returns>return order data.</returns>
+        private async Task<OrderViewModel> GetOrderDetails()
+        {
+            OrderViewModel orderFromPayment = null;
+            PaymentResponse paymentResponse = await ApiCalls.GetPaymentDetails(this.paymentId);
+
+            try
+            {
+                if (paymentResponse != null && paymentResponse.Result.Count > 0)
+                {
+                    orderFromPayment = await this.GetOrderDetails(paymentResponse.Result[0].PostBackParam.Udf1, paymentResponse.Result[0].PostBackParam.ProductInformation, paymentResponse.Result[0].PostBackParam.Udf2);
+                }
+            }
+            catch (PartnerDomainException ex)
+            {
+                this.ParsePayUException(ex);
+            }
+
+            return await Task.FromResult(orderFromPayment);
+        }
+
+        /// <summary>
+        /// get payment url.
+        /// </summary>
+        /// <param name="mode">mode of payment gateway.</param>
+        /// <returns>return string.</returns>
+        private string GetPaymentUrl(string mode)
+        {
+            ////two modes are possible sandbox and live
+            if (mode.Equals("sandbox"))
+            {
+                return Constant.TESTPAYUURL;
+            }
+
+            return Constant.LIVEPAYUURL;
+        }
+
+        /// <summary>
+        /// prepare remote post. 
+        /// </summary>
+        /// <param name="order">order details.</param>                
+        /// <param name="returnUrl">return url.</param>        
+        /// <returns>return remote post.</returns>
+        private async Task<RemotePost> PrepareRemotePost(OrderViewModel order, string returnUrl)
```

Comment:
>PrepareRemotePost [](start = 39, length = 17)

Recommend please provide better inline code documentation and such.

Reply:
Done.

---
In reply to: [183243115](https://github.com/PartnerCenterSamples/Reseller-Web-Application/pull/61#discussion_r183243115) [](ancestors = 183243115)